src/main/scala/io/moia/protos/teleproto/PbResult.scala

```diff
@@ -99,4 +105,13 @@ object PbFailure {
 
   def apply(message: String): PbFailure =
     apply("", message)
+
+  /**
+    * Collects and combines all the errors of all failures.
+    */
+  def combine(results: PbResult[_]*): PbFailure =
+    PbFailure(results.flatMap {
+      case PbSuccess(_)      => Seq.empty
```

Comment:
I don't understand how this works.
`results` is of type `PbResult`.
`flatMap` is of type `A => PbResult`.

you pass something of type `PbResult => Seq` to the `flatMap`.
Why does it even compile?